lib/pluginmanager/update.rb

```diff
@@ -52,9 +52,8 @@ def update_gems!
 
     # any errors will be logged to $stderr by invoke!
     # Bundler cannot update and clean gems in one operation so we have to call the CLI twice.
-    output = LogStash::Bundler.invoke!(:update => plugins)
+    output = LogStash::Bundler.invoke!(:update => plugins, :rubygems_source => Gem::SourceList.from(gemfile.gemset.sources))
```

Comment:
I think the `Gem::SourceList.from` should be pushed into the `LogStash::Bundler` class as a separation of concerns. the `LogStash::Bundler.invoke!` method is a logstash method and I think it does not need to know about the SourceList in its api, it will take care of transforming this structure to whatever is required.